sdk/client/src/verifying_client.rs

```diff
@@ -0,0 +1,516 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    client::Client,
+    request::MethodRequest,
+    response::{MethodResponse, Response},
+    state::State,
+};
+use anyhow::{bail, ensure, format_err, Result};
+use diem_json_rpc_types::views::AccountView;
+use diem_types::{
+    account_address::AccountAddress, account_config::diem_root_address,
+    account_state::AccountState, account_state_blob::AccountStateWithProof,
+    epoch_change::EpochChangeProof, ledger_info::LedgerInfoWithSignatures,
+    on_chain_config::RegisteredCurrencies, proof::AccumulatorConsistencyProof,
+    transaction::Version, trusted_state::TrustedState, waypoint::Waypoint,
+};
+use std::{
+    collections::HashMap,
+    convert::{TryFrom, TryInto},
+    fmt::Debug,
+    sync::{Arc, RwLock},
+};
+
+// TODO(philiphayes): figure out retry strategy
+// TODO(philiphayes): real on-disk waypoint persistence
+// TODO(philiphayes): fill out rest of the methods
+// TODO(philiphayes): use real error types
+// TODO(philiphayes): non-verifying vs verifying client equivalence testing
+// TODO(philiphayes): all clients should validate chain id (allow users to trust-on-first-use or pre-configure)
```

Comment:
So my view is that the base json-rpc client's probably shouldn't care about this, but anything wrapping it (like this verifying one) should.